app/models/manageiq/providers/ansible_tower/automation_manager/credential/api_create.rb

```diff
@@ -0,0 +1,36 @@
+class ManageIQ::Providers::AnsibleTower::AutomationManager::Credential
+  module ApiCreate
+    def create_in_provider(manager_id, params)
+      manager = ExtManagementSystem.find(manager_id)
+      credential = manager.with_provider_connection do |connection|
+        connection.api.credentials.create!(params)
+      end
+
+      # Get the record in our database
+      # TODO: This needs to be targeted refresh so it doesn't take too long
+      EmsRefresh.queue_refresh(manager, nil, true) if !manager.missing_credentials? && manager.authentication_status_ok?
```

Comment:
We just have to weigh time to implement versus current implementation.  If we already do this "wait for refresh" elsewhere, we can continue that for now, with the plan to get away from it.

@jameswnl No need to close, just split the PR.  This PR can just be the new models, and ApiCreate can be split out.